my-fund-backend/src/main/java/com/myfund/service/CategoryService.java

```diff
@@ -0,0 +1,115 @@
+package com.myfund.service;
+
+import com.myfund.model.Category;
+import com.myfund.model.DTO.CategoryDTO;
+import com.myfund.model.DTO.CreateCategoryDTO;
+import com.myfund.model.DTO.mapper.CategoryMapper;
+import com.myfund.model.DTO.mapper.SubCategoryMapper;
+import com.myfund.model.SubCategory;
+import com.myfund.model.User;
+import com.myfund.repository.CategoryRepository;
+import com.myfund.repository.SubCategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CategoryService {
```

Comment:
tests for this service pls :)

Reply:
I have created tests for the CategoryService.